File: WWF/Hands on labs 3.5/Lab03/VB/after/Exercise2/DefaultSQLServices/DefaultSQLServices/SimpleWorkflow.vb
'--------------------------------------------------------------------------------
' This file is a "Sample" as from Windows Workflow Foundation
' Hands on Labs RC
'
' Copyright (c) Microsoft Corporation. All rights reserved.
'
' This source code is intended only as a supplement to Microsoft
' Development Tools and/or on-line documentation.  See these other
' materials for detailed information regarding Microsoft code samples.
' 
' THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
' KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
' IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
' PARTICULAR PURPOSE.
'--------------------------------------------------------------------------------

'NOTE: When changing the namespace; please update XmlnsDefinitionAttribute in AssemblyInfo.vb
Public Class SimpleWorkflow
    Inherits SequentialWorkflowActivity
    Public Sub New()
        MyBase.New()
        InitializeComponent()
    End Sub

    Private Sub working_ExecuteCode(ByVal sender As System.Object, ByVal e As System.EventArgs)
        Console.WriteLine("here in working_ExecuteCode.")
    End Sub

    Private Sub timeBefore_ExecuteCode(ByVal sender As System.Object, ByVal e As System.EventArgs)
        Console.WriteLine("before delay: '{0}'", DateTime.Now.ToLongTimeString())
    End Sub

    Private Sub timeAfter_ExecuteCode(ByVal sender As System.Object, ByVal e As System.EventArgs)
        Console.WriteLine("after delay: '{0}'", DateTime.Now.ToLongTimeString())
    End Sub

    Private Sub code_ExecuteCode(ByVal sender As System.Object, ByVal e As System.EventArgs)
        Console.WriteLine("Here in Code_ExecuteCode.")
        System.Threading.Thread.Sleep(New TimeSpan(0, 0, 2))
    End Sub
End Class

File: WWF/Hands on labs 3.0/Lab03/Completed/Exercise2/DefaultSQLServices/SimpleWorkflow.cs
//--------------------------------------------------------------------------------
// This file is a "Sample" as from Windows Workflow Foundation
// Hands on Labs RC
//
// Copyright (c) Microsoft Corporation. All rights reserved.
//
// This source code is intended only as a supplement to Microsoft
// Development Tools and/or on-line documentation.  See these other
// materials for detailed information regarding Microsoft code samples.
// 
// THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
// KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
// IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//--------------------------------------------------------------------------------

using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Drawing;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;

namespace DefaultSQLServices
{
	public sealed partial class SimpleWorkflow: SequentialWorkflowActivity
	{
		public SimpleWorkflow()
		{
			InitializeComponent();
		}

        private void working_ExecuteCode(object sender, EventArgs e)
        {
            Console.WriteLine("Here in working_ExecuteCode.");
        }

        private void timeBefore_ExecuteCode(object sender, EventArgs e)
        {
            Console.WriteLine("before delay: \'{0}\'", DateTime.Now.ToLongTimeString());
        }

        private void timeAfter_ExecuteCode(object sender, EventArgs e)
        {
            Console.WriteLine("after delay: \'{0}\'", DateTime.Now.ToLongTimeString());
        }

        private void code_ExecuteCode(object sender, EventArgs e)
        {
            Console.WriteLine("Here in code_ExecuteCode.");
            System.Threading.Thread.Sleep(new TimeSpan(0, 0, 2));

        }
	}

}
